tools/src/launcher/Main.hs

```diff
@@ -200,17 +201,17 @@ serverScenario
     -> (FilePath, [Text], Maybe FilePath, Maybe FilePath)
     -- ^ Updater, args, updater runner, the update .tar
     -> Maybe String                        -- ^ Report server
-    -> Shell ()
+    -> IO ()
 serverScenario logConf node updater report = do
     runUpdater updater
     -- TODO: the updater, too, should create a log if it fails
     (_, nodeAsync, nodeLog) <- spawnNode node
     exitCode <- wait nodeAsync
-    printf ("The node has exited with "%s%"\n") (show exitCode)
+    TL.putStr $ format ("The node has exited with "%string%"\n") (show exitCode)
```

Comment:
Minor: there is `shown` formatter for this purpose. I understand that you simply changed existing code, but still :)